test_conformance/extensions/cl_ext_external_semaphore/test_external_semaphore.cpp

```diff
@@ -0,0 +1,290 @@
+    if (err != CL_SUCCESS)
+    {
+        print_error(err, "Error: Failed to get platform\n");
+        return err;
+    }
+
+    init_cl_vk_ext(platform);
+
+    VulkanDevice vkDevice;
+
+    GET_PFN(deviceID, clEnqueueSignalSemaphoresKHR);
+    GET_PFN(deviceID, clEnqueueWaitSemaphoresKHR);
+    GET_PFN(deviceID, clReleaseSemaphoreKHR);
+
+    const std::vector<VulkanExternalMemoryHandleType>
+        vkExternalMemoryHandleTypeList =
+            getSupportedVulkanExternalMemoryHandleTypeList();
+    VulkanExternalSemaphoreHandleType vkExternalSemaphoreHandleType =
+        getSupportedVulkanExternalSemaphoreHandleTypeList()[0];
+    VulkanSemaphore vkVk2CLSemaphore(vkDevice, vkExternalSemaphoreHandleType);
+
+    cl_context context2 =
+        clCreateContext(NULL, 1, &deviceID, notify_callback, NULL, &err);
+    if (!context2)
+    {
+        print_error(err, "Unable to create testing context");
+        return TEST_FAIL;
+    }
+
+    clExternalSemaphore sema1(vkVk2CLSemaphore, context,
+                              vkExternalSemaphoreHandleType, deviceID);
+    clExternalSemaphore sema2(vkVk2CLSemaphore, context2,
+                              vkExternalSemaphoreHandleType, deviceID);
+
+    clCommandQueueWrapper queue1 =
+        clCreateCommandQueue(context, deviceID, 0, &err);
+    test_error(err, "Could not create command queue");
+
+    clCommandQueueWrapper queue2 =
+        clCreateCommandQueue(context2, deviceID, 0, &err);
+    test_error(err, "Could not create command queue");
+
+    // Signal semaphore 1 and 2
+    clEventWrapper signal_event;
+    err = clEnqueueSignalSemaphoresKHR(queue1, 1, &sema1.getCLSemaphore(),
+                                       nullptr, 0, nullptr, &signal_event);
+    test_error(err, "Could not signal semaphore");
+
+    // Wait semaphore 1
+    clEventWrapper wait_1_event;
+    err = clEnqueueWaitSemaphoresKHR(queue1, 1, &sema1.getCLSemaphore(),
+                                     nullptr, 0, nullptr, &wait_1_event);
+    test_error(err, "Could not wait semaphore");
+
+    err = clEnqueueSignalSemaphoresKHR(queue2, 1, &sema2.getCLSemaphore(),
+                                       nullptr, 0, nullptr, &signal_event);
+    test_error(err, "Could not signal semaphore");
+
+    // Wait semaphore 2
+    clEventWrapper wait_2_event;
+    err = clEnqueueWaitSemaphoresKHR(queue2, 1, &sema2.getCLSemaphore(),
+                                     nullptr, 0, nullptr, &wait_2_event);
+    test_error(err, "Could not wait semaphore");
+
+    // Finish
+    err = clFinish(queue1);
+    test_error(err, "Could not finish queue");
+
+    err = clFinish(queue2);
+    test_error(err, "Could not finish queue");
+
+    // Ensure all events are completed
+    test_assert_event_complete(signal_event);
+    test_assert_event_complete(wait_1_event);
+    test_assert_event_complete(wait_2_event);
+
+    return TEST_PASS;
+}
+
+int test_external_semaphores_in_order_queue(cl_device_id deviceID,
```

Comment:
Can we reuse similar scenarios from cl_khr_semaphore by just replacing semaphore created within OpenCL by semaphore imported from vulkan?

Reply:
> Consider moving common vulkan wrapper files to a more common place say OpenCL-CTS/test_common or OpenCL-CTS/test_conformance/extensions/common. These will be needed by cl_khr_external_memory and related extensions too.

I totally agree. Do we want to do it in a scope of current task or create a new one for this?